### ISSUE TITLE
Meterpreter cannot create shell on linux - I get core_channel_write: Operation failed

### ISSUE THREAD

**Issue by drnop** [rapid7/metasploit-framework#8800]
## Steps to reproduce

Have meterpreter sessions to 2 systems, one windows and one linux Ubuntu

Windows session works fine.

On the linux session I can run commands, sysinfo, getuid, and I can upload files.

But when I try to shell I get an error on core_channel_write, see below.
This is a new issue with later versions of metasploit.

meterpreter > sysinfo
Computer     : iot.notmyname.com
OS           : Ubuntu 10.10 (Linux 2.6.35-22-server)
Architecture : x64
Meterpreter  : x86/linux
meterpreter > getuid
Server username: uid=33, gid=33, euid=33, egid=33
meterpreter > shell
Process 1130 created.
Channel 3 created.

[-] core_channel_write: Operation failed: 9
meterpreter >

### Metasploit version

msf exploit(apache_mod_cgi_bash_env_exec) > version
Framework: 4.15.2-dev
Console  : 4.15.2-dev

Get this with the `version` command in msfconsole (or `git log -1 --pretty=oneline` for a source install).

### I installed Metasploit with:
- [ ] Kali package via apt

### OS

What OS are you running Metasploit on?

root@kali2:~# cat /etc/issue
Kali GNU/Linux Rolling \n \l

**Comment by sn0wfa11** [comment]
Which exact payload are you using for the Linux machine? 

I am finding this problem with the below Linux based payloads.  I am not having this issue with any of the staged, non-http(s) based payloads.

```
linux/armle/meterpreter_reverse_https
linux/mipsbe/meterpreter_reverse_https
linux/x86/meterpreter_reverse_https
linux/x64/meterpreter_reverse_https
```

**Comment by drnop** [comment]
I am using linux/x86/meterpreter/bind_tcp.
This worked in older versions of metasploit (in exactly the same lab environment).

**Comment by busterb** [comment]
When you say older, how much older? A year ago?

**Comment by drnop** [comment]
Oops. I find the issue.  I used /x86/meterpreter/bind_tcp but the target is 64 bit system. When I changed to x64/meterpreter/bind_tcp the shell worked.
Funny in that the older version of metasploit (built into kali version 2, so a bit more than a year old) the x86 version worked too.
Sorry about the mistake, since everything was scripted and used to work I did not check the old scripts (that used wrong payload).
